Solutions/Marain.ContentManagement.Rendering.Abstractions/Marain/Cms/IContentRenderer.cs

```diff
@@ -0,0 +1,26 @@
+﻿// <copyright file="IContentRenderer.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Cms
+{
+    using System.IO;
+    using System.Threading.Tasks;
+    using Corvus.Extensions.Json;
+
+    /// <summary>
+    /// Implemented by types that render an <see cref="IContentPayload"/>.
+    /// </summary>
+    public interface IContentRenderer
+    {
+        /// <summary>
+        /// Render the provided content to the output stream.
+        /// </summary>
+        /// <param name="output">The output text writer to which to render the content.</param>
+        /// <param name="parentContent">The root content element which contains this payload.</param>
+        /// <param name="currentPayload">The current payload to be rendered.</param>
+        /// <param name="context">The context property bag.</param>
+        /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+        public Task RenderAsync(TextWriter output, Content parentContent, IContentPayload currentPayload, PropertyBag context);
```

Comment:
Well; this is a can of worms.

We are using PropertyBag because it delays typing (and deserialization) until you pull the content out of the bag, and we are using Json as the underlying storage mechanism for that.

In real applications, this will come from a JSON payload passed through a request, and into this bag; we have no real idea what it will contain until the point it is consumed by a specific renderer. This is a nice abstraction of this particular problem that is used elsewhere (e.g. in the `Content` itself which also has such a loosely typed property bag) so we already have this dependency in the tree anyway.

There is an argument to say that we should move this property bag out of Corvus.Extensions.Newtonsoft.Json and into its own thing, though. But that is a different problem.